File: domain-driven-design/go-web/internal/middleware/logging.go
package middleware

import (
  "go-order-system/pkg/utils"
  "net/http"
  "time"
)

// LoggingMiddleware 记录请求日志的中间件
func LoggingMiddleware(next http.HandlerFunc) http.HandlerFunc {
  return func(w http.ResponseWriter, r *http.Request) {
    start := time.Now()
    // 继续处理请求
    next(w, r)
    // 记录日志
    // log.Printf("%s %s took %v", r.Method, r.URL.Path, time.Since(start))
    utils.LogPrintf(r, start)
  }
}

File: domain-driven-design/nodejs-web/src/middleware/logging.js
// logging_middleware.js
function loggingMiddleware(req, res, next) {
  const start = Date.now();
  res.on('finish', () => {
    const duration = Date.now() - start;
    console.log(`REQUEST: ${req.method} ${req.url} took ${duration}ms`);
  });
  next();
}

module.exports = { loggingMiddleware };